File: jrc_abogados.client/src/app/Models/Documento.ts
import { Cliente } from "./Cliente";
import { Expediente } from "./Expediente";

export class Documento {
  id: number = 0;
  nombre: string = "";
  path: string = "";
  fechaInicio: string = "";
  descripcion: string = "";
  clienteId = "";
  cliente: Cliente = new Cliente;
  expedienteId = 0;
  expediente: Expediente = new Expediente;
  empleadoId: number = 0;
  empleado: any;
}

File: JRC_Abogados.Server/Models/Documento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JRC_Abogados.Server.Models
{
    public class Documento
    {
        [Key]
        public int Id { get; set; }
        public string Nombre { get; set; }
        public string? Descripcion { get; set; }
        public string Path { get; set; }
        public DateTime FechaInicio { get; set; }
        public int ExpedienteId { get; set; }
        [ForeignKey("ExpedienteId")]
        public virtual Expediente Expediente { get; set; }
        public int ClienteId { get; set; }
        [ForeignKey("ClienteId")]
        public virtual Cliente? Cliente { get; set; }
        public int EmpleadoId { get; set; }
        [ForeignKey("EmpleadoId")]
        public virtual Empleado? Empleado { get; set; }
    }
}
